tools/ReleaseNotes/release_editor.py

```diff
@@ -6,15 +6,17 @@
 # SPDX - License - Identifier: GPL - 3.0 +
 
 import os
+import pathlib
 import sys
-import shutil
+
 
 DIRECTIVE = ".. amalgamate:: "
 
 
-def getReleaseRoot():
-    script_dir = os.path.split(sys.argv[0])[0]
-    return os.path.abspath(os.path.join(script_dir, '../../docs/source/release/'))
+def getReleaseRoot() -> pathlib.Path:
```

Comment:
I just realised when testing this out that this has a built-in assumption that the current directory is the same as this script. I know the old script did this but it's a good habit to get into to try and make it so you can be in any directory, i.e. this failed for me

```sh
$ python tools/ReleaseNotes/release_editor.py --release 6.4.0
```

with the following error:

```sh
Traceback (most recent call last):
  File "tools/ReleaseNotes/release_editor.py", line 123, in <module>
    directoriesUsed = checkContainsReleaseNote(mainDirectory)
  File "tools/ReleaseNotes/release_editor.py", line 87, in checkContainsReleaseNote
    getReleaseNoteDirectories(path, releaseNoteList)
  File "tools/ReleaseNotes/release_editor.py", line 73, in getReleaseNoteDirectories
    if len([f for f in os.listdir(path) if f.endswith('.rst')]) > 0:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/dmn58364/Code/git/docs/source/release/v6.4.0'
```

Could we swap out using `sys.argv` for `__file__. The latter variable will always contain a full path to this file wherever it is on the system. We can then be confident of the relative path of the other directories.